classes/manager.php

```diff
@@ -118,58 +120,55 @@ public static function sample_period(): float {
      * @throws \dml_exception
      */
     public static function init(): void {
-        $sampleperiod = self::sample_period();
+        self::$profiler = new \stdClass();
```

Comment:
Why did these need to become static. Could this lead to any weird collisions with concurrent single process profiling?